Mathlib/Analysis/CStarAlgebra/ContinuousFunctionalCalculus/NonUnital.lean

```diff
@@ -648,6 +648,50 @@ end Ring
 
 end Order
 
+/-! ### `cfcₙHom` on a superset of the quasispectrum -/
+
+section Superset
+
+open ContinuousMapZero
+
+variable {R A : Type*} {p : A → Prop} [CommSemiring R] [Nontrivial R] [StarRing R]
+    [MetricSpace R] [TopologicalSemiring R] [ContinuousStar R] [NonUnitalRing A] [StarRing A]
+    [TopologicalSpace A] [Module R A] [IsScalarTower R A A] [SMulCommClass R A A]
+    [instCFCₙ : NonUnitalContinuousFunctionalCalculus R p]
+
+/-- The composition of `cfcₙHom` with the natural embedding `C(s, R)₀ → C(quasispectrum R a, R)₀`
+whenever `quasispectrum R a ⊆ s`.
+
+This is sometimes necessary in order to consider the same continuous functions applied to multiple
+distinct elements, with the added constraint that `cfcₙ` does not suffice. This can occur, for
+example, if it is necessary to use uniqueness of this continuous functional calculus. A practical
+example can be found in the proof of `CFC.posPart_negPart_unique`. -/
+@[simps!]
+noncomputable def cfcₙHomSuperset {a : A} (ha : p a) {s : Set R} (hs : σₙ R a ⊆ s) :
+    letI : Zero s := ⟨0, hs (quasispectrum.zero_mem R a)⟩
+    C(s, R)₀ →⋆ₙₐ[R] A :=
```

Comment:
it has a different meaning from Lean 3. In Lean 3 it reset the instance cache since that was frozen. In Lean 4, we don't freeze the instance cache anymore, so that's not relevant, but now the `I` means "inline". I thought it would be better to inline this, but maybe it doesn't matter. I'll check before merging.

Reply:
Ah I see! Yeah I forgot what it did in Lean 4; inlining certainly makes sense here.